File: 5. Concatenation of Array/Two Pass/concat-tp.ts
const getConcatenation = (nums: number[]) => {
  const n = nums.length;
  const ans: number[] = [];
  for (let i = 0; i < 2; i++) {
    for (const num of nums) {
      ans.push(num);
    }
  }
  return ans;
};

const nums = [1, 2, 3, 5];
const result = getConcatenation(nums);
console.log(`Concatenated Array: ${result}`);

File: 5. Concatenation of Array/Two Pass/concat-tp.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
  public:
    vector<int> getConcatenation(vector<int>& nums) {
      vector<int> ans;
      for (int i = 0; i < 2; ++i) {
        for (int num: nums) {
          ans.push_back(num);
        }
      }
      return ans;
    }
};

int main() {
  Solution s1;
  vector<int> nums = {1, 2, 4, 3};
  vector<int> result = s1.getConcatenation(nums);
  cout << "Concatenated array: ";
  for (int num: result) {
    cout << num << " ";
  }
  cout << endl;
  return 0;
}
